### PR TITLE
ApiXMLRPC: Fixed return value for local mode

### PR DESCRIPTION
**Pull request by CeltonMcGrath**

### CODE REVIEW

File: src/tactic/command/watch_drop_folder.py [hunk 1]
```diff
@@ -113,15 +114,23 @@ def _run(my):
...
             error_path = "%s.error" % path
 
+            if not os.path.exists(path):
+                continue
             if not os.path.exists(checkin_path):
-                print "ERROR: no checkin path [%s]" % checkin_path
+                print "Action Thread SKIP: no checkin path [%s]" % checkin_path
                 continue
+            else:
+                f = open(checkin_path, "r")
```
Comment (CeltonMcGrath) [inline src/tactic/command/watch_drop_folder.py]: Add a comment

File: src/pyasm/common/common.py [hunk 2]
```diff
@@ -1241,17 +1241,24 @@ def is_exe(fpath):
...
         '''Kills the current program.'''
+        
         import sys
         if not pid:
             pid = os.getpid()
         pid = int(pid)
+        
+        log_path = "C:/pid_%s.txt" % pid
+        f.open(log_path, "w")
+        f.close()
```
Comment (oboreo) [inline src/pyasm/common/common.py]: remove this

File: src/tactic/command/watch_drop_folder.py [hunk 3]
```diff
@@ -160,15 +172,34 @@ def _run(my):
...
                 checkin_path = "%s.checkin" % path
+                lock_path = "%s.lock" % path
                 if os.path.exists(checkin_path):
                     os.unlink(checkin_path)
-            Common.restart()
+                if os.path.exists(lock_path):
+                    os.unlink(lock_path)
+            # this exaggerates the effect of not pausing check thread for cleaning
+            #time.sleep(10)
+            print "\n\nrestarting now!!"
```
Comment (oboreo) [inline src/tactic/command/watch_drop_folder.py]: remove print  and change the line to Common.kill()

File: src/pyasm/web/monitor.py [hunk 4]
```diff
@@ -531,6 +528,17 @@ def set_check_interval(my, check_interval):
     def set_dev(my, mode):
         my.dev_mode = mode
 
+    def watch_folder_cleanup(my, base_dir):
+        '''removes old action files from prevous watch
```
Comment (oboreo) [inline src/pyasm/web/monitor.py]: previous

File: src/tactic/command/watch_drop_folder.py [hunk 5]
```diff
@@ -187,26 +216,43 @@ def run(my):
...
                 return
 
+            task = my.kwargs.get("task")
+ 
+            if task.in_clean():
+                return
+            
+            pid = os.getpid()
             f = open(my.lock_path, "w")
+            #f.write(str(pid))
```
Comment (oboreo) [inline src/tactic/command/watch_drop_folder.py]: remove commented out section

File: src/tactic/command/watch_drop_folder.py [hunk 6]
```diff
@@ -187,26 +216,43 @@ def run(my):
...
             f.close()
 
             changed = my.verify_file_size(path)
-
-
+            # Return if another process has locked this file last
```
Comment (oboreo) [inline src/tactic/command/watch_drop_folder.py]: remove commented out section

File: src/pyasm/web/monitor.py [hunk 7]
```diff
@@ -531,6 +531,17 @@ def set_check_interval(my, check_interval):
     def set_dev(my, mode):
         my.dev_mode = mode
 
+    def watch_folder_cleanup(my, base_dir):
+        '''removes old action files from previous watch
+        folder processes.'''
+        files = os.listdir(base_dir)
+        for file in files:
```
Comment (oboreo) [inline src/pyasm/web/monitor.py]: avoid using keyword or method name like file as variable

File: src/tactic/command/watch_drop_folder.py [hunk 8]
```diff
@@ -92,33 +92,44 @@ def run(my):
...
-
             checkin_path = "%s.checkin" % path
+            lock_path = "%s.lock" % path
             error_path = "%s.error" % path
 
+            if not os.path.exists(path):
+                continue
             if not os.path.exists(checkin_path):
-                print "ERROR: no lock path [%s]" % checkin_path
+                print "Action Thread SKIP: no checkin path [%s]" % checkin_path
```
Comment (oboreo) [inline src/tactic/command/watch_drop_folder.py]: this print statement is good for debugging.. comment it out.

File: src/tactic/command/watch_drop_folder.py [hunk 9]
```diff
@@ -683,7 +679,7 @@ def start(cls):
         parser.add_option("-s", "--search_type", dest="search_type", help="Define search_type.")
         parser.add_option("-P", "--process", dest="process", help="Define process.")
         parser.add_option("-S", "--script_path",dest="script_path", help="Define script_path.")
-
+        parser.add_option("-C", "--code",dest="code", help="Define watch folder code.")
```
Comment (oboreo) [inline src/tactic/command/watch_drop_folder.py]: should rename it as -w --watch_folder_code

File: src/tactic/command/watch_drop_folder.py [hunk 10]
```diff
@@ -734,6 +730,18 @@ def start(cls):
         else:
             handler = None
 
+        if options.code != None:
```
Comment (oboreo) [inline src/tactic/command/watch_drop_folder.py]: rename all variables code to watch_folder_code

File: src/tactic/command/watch_drop_folder.py [hunk 11]
```diff
@@ -661,11 +725,32 @@ def start(cls):
...
+            watch_folder_code = None
+        if watch_folder_code:   
+            # record pid in watch folder pid file
+            pid = os.getpid()
+            pid_file = "%s/log/watch_folder.%s" % (Environment.get_tmp_dir(), watch_folder_code)
+            f = open(pid_file, "w")
+            f.write(str(pid))
+            f.close()
+
         Batch(project_code=project_code, site=site)
```
Comment (oboreo) [inline src/tactic/command/watch_drop_folder.py]: That's good. more readable

**Pull request merged**